File: app/src/main/java/eu/caraus/dynamo/application/common/schedulers/TestSchedulerProvider.java
package eu.caraus.dynamo.application.common.schedulers;

import io.reactivex.Scheduler;
import io.reactivex.schedulers.Schedulers;

public class TestSchedulerProvider implements SchedulerProvider {

    @Override
    public Scheduler ui() {
        return Schedulers.trampoline();
    }

    @Override
    public Scheduler io() {
        return Schedulers.trampoline();
    }

    @Override
    public Scheduler computation() {
        return Schedulers.trampoline();
    }

}

File: app/src/main/java/eu/caraus/dynamo/application/common/schedulers/TestSchedulerProvider.kt
package eu.caraus.dynamo.application.common.schedulers

import io.reactivex.Scheduler
import io.reactivex.schedulers.Schedulers

class TestSchedulerProvider : SchedulerProvider {

    override fun ui()           : Scheduler {
        return Schedulers.trampoline()
    }

    override fun io()           : Scheduler {
        return Schedulers.trampoline()
    }

    override fun computation()  : Scheduler {
        return Schedulers.trampoline()
    }

}
